data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/GetProposerDuties.java

```diff
@@ -0,0 +1,130 @@
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.RES_INTERNAL_ERROR;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.RES_OK;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.RES_SERVICE_UNAVAILABLE;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.TG_V1_VALIDATOR;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+import tech.pegasys.teku.api.ChainDataProvider;
+import tech.pegasys.teku.api.DataProvider;
+import tech.pegasys.teku.api.SyncDataProvider;
+import tech.pegasys.teku.api.ValidatorDataProvider;
+import tech.pegasys.teku.api.response.v1.validator.GetProposerDutiesResponse;
+import tech.pegasys.teku.api.response.v1.validator.ProposerDuty;
+import tech.pegasys.teku.beaconrestapi.handlers.AbstractHandler;
+import tech.pegasys.teku.beaconrestapi.schema.BadRequest;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.provider.JsonProvider;
+
+public class GetProposerDuties extends AbstractHandler implements Handler {
+  private static final Logger LOG = LogManager.getLogger();
+  public static final String ROUTE = "/eth/v1/validator/duties/proposer/:epoch";
+  private final ValidatorDataProvider validatorDataProvider;
+  private final SyncDataProvider syncDataProvider;
+  private final ChainDataProvider chainDataProvider;
+
+  public GetProposerDuties(final DataProvider dataProvider, final JsonProvider jsonProvider) {
+    super(jsonProvider);
+    this.validatorDataProvider = dataProvider.getValidatorDataProvider();
+    this.syncDataProvider = dataProvider.getSyncDataProvider();
+    this.chainDataProvider = dataProvider.getChainDataProvider();
+  }
+
+  GetProposerDuties(
+      final ChainDataProvider chainDataProvider,
+      final SyncDataProvider syncDataProvider,
+      final ValidatorDataProvider validatorDataProvider,
+      final JsonProvider jsonProvider) {
+    super(jsonProvider);
+    this.chainDataProvider = chainDataProvider;
+    this.validatorDataProvider = validatorDataProvider;
+    this.syncDataProvider = syncDataProvider;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Get proposer duties",
+      tags = {TG_V1_VALIDATOR},
+      description =
+          "Request beacon node to provide all validators that are supposed to propose a block in the given epoch.",
```

Comment:
Unless this came from the standard spec I'd suggest:
```suggestion
          "Request beacon node to provide all validators that are scheduled to propose a block in the given epoch.",
```

Reply:
it was a cut and paste, but i still think i'll change it. might raise a PR to fix the api version too.